### DISCUSSION TITLE
Snap Zone Initial or default object

### DISCUSSION THREAD

**Discussion by agupta2000** [ExtendRealityLtd/Tilia.Interactions.SnapZone.Unity#98] [General]
Hello, how do you set the default or initial object in the snap zone?  I spent hours and can't figure it out.  Please help.

**Comment by thestonefox** [comment]
To set an interactable object to be the default snapped object use a Behaviour Enabled Observer component:

1. Add Snap Zone to scene
2. Add Interactable object to scene
3. Add a Behaviour Enabled Observer via the Zinnia menu option `Main Menu -> Window -> Zinnia -> Observable List Component Generator`
![image](https://user-images.githubusercontent.com/1029673/112138280-3e9b8900-8bc9-11eb-9968-265903e72d9c.png)
4. In the Elements list on the Behaviour Enabled Observer, listen for the Snap Zone and the Interactable to be snapped to be enabled then in the `Active And Enabled` event call the `Snap` method on the `SnapZoneFacade` passing in the `Interactable` you want to snap by default.
![image](https://user-images.githubusercontent.com/1029673/112138517-815d6100-8bc9-11eb-8a83-4625c8c8a15c.png)


This will now snap an object to the snap zone when the snap zone and the object are both enabled.

**Comment by agupta2000** [comment]
Thank you. This worked for me.

On Mar 23, 2021, at 7:19 AM, The Stonefox ***@***.***> wrote:

﻿

To set an interactable object to be the default snapped object use a Behaviour Enabled Observer component:

  1.  Add Snap Zone to scene
  2.  Add Interactable object to scene
  3.  Add a Behaviour Enabled Observer via the Zinnia menu option Main Menu -> Window -> Zinnia -> Observable List Component Generator
[image]<https://user-images.githubusercontent.com/1029673/112138280-3e9b8900-8bc9-11eb-9968-265903e72d9c.png>
  4.  In the Elements list on the Behaviour Enabled Observer, listen for the Snap Zone and the Interactable to be snapped to be enabled then in the Active And Enabled event call the Snap method on the SnapZoneFacade passing in the Interactable you want to snap by default.
[image]<https://user-images.githubusercontent.com/1029673/112138517-815d6100-8bc9-11eb-8a83-4625c8c8a15c.png>

This will now snap an object to the snap zone when the snap zone and the object are both enabled.
